From c2d90128a3805ac6127d49d8542846d757aee70e Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 18 Sep 2018 06:01:43 +0300
Subject: [PATCH] EEprom Layout: update to new format

Remove Infineon field since it can be read directly from device
Add Max power/Freq fields for our power management
Add Fans for our thermal management
---

diff --git a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
index 50d17f0..57c5195 100755
--- a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
+++ b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
@@ -39,10 +39,12 @@
 	print_field_str "Production Serial Number" 1 11
 	print_field_str "PCB Assembly Version" 12 4
 	print_field_str "PCB Version" 16 4
-	print_field_num "Infineon FW Version" 20 1
-	print_field_num "DDR Type and Speed" 21 1
-	print_field_num "PCIe Gen3 or Gen4 Enabled" 22 1
-	print_field_str "Board Configuration" 23 64
+	print_field_num "DDR Type and Speed" 20 1
+	print_field_num "PCIe Gen3 or Gen4 Enabled" 21 1
+	print_field_num "Max Frequency [MHz]" 22 2
+	print_field_num "Max Power [Watt]" 24 1
+	print_field_num "Fans" 25 1
+	print_field_str "Board Configuration" 26 64
 else
 	echo "Unknown eeprom layout"
 fi
diff --git a/board/habanalabs/goya/uboot-eeprom.cmd b/board/habanalabs/goya/uboot-eeprom.cmd
index 7f82368..4a64c95 100644
--- a/board/habanalabs/goya/uboot-eeprom.cmd
+++ b/board/habanalabs/goya/uboot-eeprom.cmd
@@ -5,9 +5,11 @@
 eeprom update $LAYOUT "Production Serial Number" "AI11223344"
 eeprom update $LAYOUT "PCB Assembly Version" "V1A"
 eeprom update $LAYOUT "PCB Version" "R0B"
-eeprom update $LAYOUT "Infineon FW Version" "3c"
 eeprom update $LAYOUT "DDR Type and Speed" "12"
 eeprom update $LAYOUT "PCIe Gen3 or Gen4 Enabled" "3"
+eeprom update $LAYOUT "Max Frequency [MHz]" "064a"
+eeprom update $LAYOUT "Max Power [Watt]" "c8"
+eeprom update $LAYOUT "Fans" "2"
 eeprom update $LAYOUT "Board Configuration" "Goya HL1000"
 echo "eeprom content is:"
 eeprom print  $LAYOUT
